### ISSUE TITLE
Correctly process nemesis failures in parallel nemesis runs

### ISSUE THREAD

**Issue by aleksbykov** [scylladb/scylla-cluster-tests#6618]
### Prerequisites

* [x] Are you rebased to master ?
* [ ] Is it reproducible ?
* [ ] Did you perform a cursory search if this issue isn't opened ?

### Versions
* **SCT**: master
* **scylla**: master

### Logs
* test_id: d14684e9-9cd9-430d-8962-10888251f1ce
* job log: https://cloudius-jenkins-test.s3.amazonaws.com/d14684e9-9cd9-430d-8962-10888251f1ce/20230909_052616/sct-d14684e9.log.tar.gz
* Argus: https://argus.scylladb.com/test/024e78ba-3671-4fd0-bafc-b48b2a978f59/runs?additionalRuns[]=d14684e9-9cd9-430d-8962-10888251f1ce

### Description
The job run in parallel 2 nemesis threads: Disruption nemesis and schema changes nemesis

If nemesises run in parallel we have cluster health checker disabled. During provided job, there are a lot of connection errors. and these 2 cases cause the job failed. The first problem happened, when Decommission nemesis start running on node6, but due to connection error, the command execution was termanted:
```
  File "/home/ubuntu/scylla-cluster-tests/sdcm/remote/remote_libssh_cmd_runner.py", line 78, in _run_on_retryable_exception
    raise RetryableNetworkException(str(exc), original=exc)
sdcm.remote.base.RetryableNetworkException
```
and the whole nemesis was stopped. But decommission process on node was running and finished successfully. But node stay in sct cluster list. So when next nemesis terminate and relplace is starting, the nemesis terminate the node, but verification node was randomly choosen node6 where decommission was running, and nodetool status return , that node5 is still up and this cause abort the nemesis with error:
```
  File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 1487, in disrupt_terminate_and_replace_node
    assert get_node_state(old_node_ip) == "DN", "Removed node state should be DN"
AssertionError: Removed node state should be DN
```
and again new node was not added. And at some point we lost quorum and cluster stability, which cause next nemesis also failed

We need some workaround for that situations:
1. Enable health checker, but this could bring new issue to sct stability
2. Raft topology in future should resolve such issues, but for now may be we need add some wrapper for disruption nemesis, which allow to check cluster status before run next disrupt nemesis in same thread
3. add addtitional checks to decommission nemesis, if command failed, check node state, connect and approve that decommission is still running and wait while it finished.

**Comment by roydahan** [comment]
This is not fixable in any way at the moment without huge investment which still may have many edge cases.

The current possible solution is just to ignore "Error events" during the test itself and to only run "teardown validator" which can be: healthcheck, no corrdumps, sstables (using scrub), etc.
